worker/task_executor.go

```diff
@@ -0,0 +1,164 @@
+package worker
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	TaskScriptExecutorFileName  = "task-script-executor.sh"
+	RunTaskScriptCommandPattern = "/bin/bash %s -d -tsf=%s"
+	DefaultWorkerDockerImage    = "ubuntu"
+)
+
+type TaskExecutor struct {
+	Cli client.Client
+	Cid string
+}
+
+func (e *TaskExecutor) Execute(task *Task, containerSpawnWarner chan<- interface{}) error {
+	image := task.Image
+
+	if image == "" {
```

Comment:
Uhm ... should we have a default image?

Reply:
we do that in the next line..

Reply:
sorry, my suggestion was poorly written .. my point is that I see no advantage of having the defaultDockerImage. In my opinion, it should be a mandatory parameter. What do you thing?

Reply:
that's an option, i'll return an error in case of empty image then

Reply:
what happens if we allow the empty string to be passed down the docker util module? If the docker util already raises an error, I'd move this check up to the place where we build the Task type